### PR TITLE
Set name for the main thread

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

The main thread will be displayed as `ClickHouse` instead of `clickhouse-serv`.

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 94af9e8fc7629e78f7ad4562e67ebd68e436502a with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53600/94af9e8fc7629e78f7ad4562e67ebd68e436502a/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by alexey-milovidov** [comment]
Need to check how it looks in `top`, it should be displayed as before.

**Comment by alexey-milovidov** [comment]
`top` displays it.

Let's change to `ClickHouse` instead.

File: programs/server/Server.cpp [hunk 1]
```diff
@@ -575,6 +575,7 @@ static void sanityChecks(Server & server)
 int Server::main(const std::vector<std::string> & /*args*/)
 try
 {
+    setThreadName("ClickHouse");
```
Comment (devcrafter) [inline programs/server/Server.cpp:578]: Did you want to set the main thread name to "Main"? (as in description to the PR)

Comment (devcrafter) [reply programs/server/Server.cpp:578]: Ah, did read all your comments.

**Review by devcrafter — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
It will break the usages of `pidof` and `killall`, let's not do that.

**Comment by alexey-milovidov** [comment]
`pidof` works but `killall` does not:

```
milovidov@milovidov-desktop:~/work/ClickHouse$ pidof clickhouse-server
114379
milovidov@milovidov-desktop:~/work/ClickHouse$ killall clickhouse-server
clickhouse-server: no process found
milovidov@milovidov-desktop:~/work/ClickHouse$ killall clickhouse-serv
clickhouse-serv: no process found
```

**Pull request closed without merging**